apps/src/studio/studio.js

```diff
@@ -1183,17 +1183,19 @@ function edgeCollidableCollisionDistance (collidable, edgeName, yAxis) {
  */
 function handleActorCollisionsWithCollidableList (
            spriteIndex, xCenter, yCenter, list, autoDisappear) {
-  // Traverse the list in reverse order because we may remove elements from the
-  // list while inside the loop:
-  for (var i = list.length - 1; i >= 0; i--) {
-    var collidable = list[i];
-    var next = collidable.getNextPosition();
+  var collidable, next, className, numItemsOfClass;
+  var distanceScaling = constants.SPRITE_COLLIDE_DISTANCE_SCALING;
 
-    if (collidable.isFading && collidable.isFading()) {
-      continue;
```

Comment:
Here, you'll notice we do one thing right - we skip items that are already fading out while we loop through items checking collisions.  This makes sure you can't collect an item that's already fading out.